### PR TITLE
RESTful controller

### PR DESCRIPTION
**Pull request by mostafa-221**

### CODE REVIEW

**Comment by mostafa-221** [comment]
Marked as a draft for missing unit tests

**Review by tkoen93 — CHANGES_REQUESTED** [review]
Tried to see if this feature branch is compatible with the current frontend. Looks allright, but when trying to relink skills to the vacancies it seems that this functionality doesn't work properly in the backend. It doesn't link anything. Can you look into this if you've got time @mostafa-221 ? Otherwise, let me know and I'll look into it myself.

File: src/main/java/nl/ordina/jobcrawler/controller/SkillMatcherController.java [hunk 1]
```diff
@@ -14,22 +13,18 @@
...
-
     // rematch the skills
-    @PutMapping(path = "skillmatcher")
-    public ResponseEntity<ResponseCode> relinkSkills() {
+    @PutMapping(path = "/skillmatcher")
+    public void relinkSkills() {
         log.info("relink skills");
         Thread newThread = new Thread(skillMatcherService::relinkSkills);
         newThread.start();
-        return new ResponseEntity<>(new ResponseCode("OK"), HttpStatus.OK);
```
Comment (mostafa-221) [inline src/main/java/nl/ordina/jobcrawler/controller/SkillMatcherController.java]: This method was changed from returning this body 
```
{
  "errorCode": "OK"
}
```
on which the frontend verifies if the function ran correctly, to returning no body as this is a more standard way. 
I think it is better to adjust the frontend to this.

Comment (tkoen93) [reply src/main/java/nl/ordina/jobcrawler/controller/SkillMatcherController.java]: The frontend actually doesn't do anything with the response (I do get an error in console, but the put request is still being processed by the backend). But judging by the logs in IntelliJ and looking into the vacancy_skills table I don't see any link being made.

Comment (mostafa-221) [reply src/main/java/nl/ordina/jobcrawler/controller/SkillMatcherController.java]: Just tested it and the problem is the findByOrderByNameAsc which was not included in the this controller. I am working on adding it.

File: src/main/java/nl/ordina/jobcrawler/model/assembler/SkillModelAssembler.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package nl.ordina.jobcrawler.model.assembler;
+
```
Comment (adriaanbakker-java) [inline src/main/java/nl/ordina/jobcrawler/model/assembler/SkillModelAssembler.java:2]: please add some comment - what is the purpose of the SkillModelAssembler? What is a RepresentationModelAssembler and how is this beneficial to the skill model?

File: src/main/java/nl/ordina/jobcrawler/repo/SkillRepository.java [hunk 3]
```diff
@@ -1,23 +1,16 @@
...
-import org.springframework.web.bind.annotation.CrossOrigin;
 
 import java.util.List;
-import java.util.Optional;
-import java.util.Set;
 import java.util.UUID;
 
 @Repository
-@RepositoryRestResource
-@CrossOrigin
```
Comment (adriaanbakker-java) [inline src/main/java/nl/ordina/jobcrawler/repo/SkillRepository.java:20]: what happened to the crossorigin, do we not do that at the moment?

File: src/main/java/nl/ordina/jobcrawler/repo/VacancyRepository.java [hunk 4]
```diff
@@ -6,18 +6,14 @@
...
 import org.springframework.stereotype.Repository;
-import org.springframework.web.bind.annotation.CrossOrigin;
 
 import javax.transaction.Transactional;
 import java.util.Optional;
 import java.util.Set;
 import java.util.UUID;
 
 @Repository
-@RepositoryRestResource
```
Comment (adriaanbakker-java) [inline src/main/java/nl/ordina/jobcrawler/repo/VacancyRepository.java:19]: is it not a repositoryRestResource anymore? Why?

**Review by adriaanbakker-java — CHANGES_REQUESTED** [review]
please put some comment with the main changes for the annotations

**Review by tkoen93 — APPROVED** [review]

**Review by adriaanbakker-java — APPROVED** [review]

File: src/main/java/nl/ordina/jobcrawler/repo/SkillRepository.java (hunk 3 above)
Comment (mostafa-221) [reply src/main/java/nl/ordina/jobcrawler/repo/SkillRepository.java:20]: it was moved to our own implementation class, the SkillController class

File: src/main/java/nl/ordina/jobcrawler/repo/VacancyRepository.java (hunk 4 above)
Comment (mostafa-221) [reply src/main/java/nl/ordina/jobcrawler/repo/VacancyRepository.java:19]: Because we decided we will be writing our own implementation

Comment (mostafa-221) [reply src/main/java/nl/ordina/jobcrawler/repo/VacancyRepository.java:19]: instead of letting this annotation do it for us

**Pull request merged**